lib/sparql_to_sw_solr/instance_solr_doc.rb

```diff
@@ -25,14 +33,62 @@ def solr_doc_hash
         return if CKEY_BLACKLIST.include?(@ckey)
         doc = {}
         doc[:id] = instance_uri_to_ckey
-        doc
+        build_doc_titles(doc)
       end
     end
 
     private
 
+    def build_doc_titles(doc)
+      doc[:id] = instance_uri_to_ckey
+      doc[:title_245a_search] = primary_triples(['mainTitle'], title_query_str)
+      doc[:title_245_search] = primary_triples(['rdf-schema#label'], title_query_str)
+      doc[:title_245a_display] = primary_triples(['mainTitle'], title_query_str)
+      doc[:title_display] = primary_triples(%w(mainTitle subtitle), title_query_str)
+      doc[:title_full_display] = "#{primary_triples(%w(mainTitle subtitle), title_query_str)} / " \
+                                 "#{primary_triples(['responsibilityStatement'], responsibility_query_str)}"
+      doc
+    end
+
     def instance_uri_to_ckey
       @ckey ||= self.class.instance_uri_to_ckey(@instance_uri)
     end
+
+    def sparql
+      @sparql ||= SPARQL::Client.new(SPARQL_URL)
+    end
+
+    def title_query_str
```

Comment:
I'm not a huge fan of constants masquerading as methods - can we set the constant in the initializer, if we need instance_uri defined?.  Also, this is perhaps more accurately named `PRIMARY_TITLE_SPARQL` or something indicating it is about the primary title, no?